src/main/java/org/folio/service/orders/utils/ProductIdUtils.java

```diff
@@ -0,0 +1,84 @@
+package org.folio.service.orders.utils;
+
+import org.apache.commons.lang3.StringUtils;
+import org.folio.rest.jaxrs.model.CompositePoLine;
+import org.folio.rest.jaxrs.model.ProductId;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+import static java.util.stream.Collectors.groupingBy;
+import static java.util.stream.Collectors.toList;
+
+public class ProductIdUtils {
+
+  public static Set<String> buildSetOfProductIdsFromCompositePoLines(List<CompositePoLine> compositePoLines, String isbnTypeId) {
+    List<ProductId> productIds = compositePoLines.stream()
+      .flatMap(pol -> pol.getDetails().getProductIds().stream())
+      .collect(Collectors.toList());
+    return buildSetOfProductIds(productIds, isbnTypeId);
+  }
+
+  public static Set<String> buildSetOfProductIds(List<ProductId> productIds, String isbnTypeId) {
+    return productIds.stream()
+      .filter(productId -> isISBN(isbnTypeId, productId))
+      .map(ProductId::getProductId)
+      .collect(Collectors.toSet());
+  }
+
+  public static boolean isISBN(String isbnTypeId, ProductId productId) {
+    return Objects.equals(productId.getProductIdType(), isbnTypeId);
+  }
+
+  public static void removeISBNDuplicates(CompositePoLine compPOL, String isbnTypeId) {
+    List<ProductId> productIds = removeISBNDuplicates(compPOL.getDetails().getProductIds(), isbnTypeId);
+    compPOL.getDetails().setProductIds(productIds);
+  }
+
+  public static List<ProductId> removeISBNDuplicates(List<ProductId> productIds, String isbnTypeId) {
+    List<ProductId> notISBNs = getNonISBNProductIds(productIds, isbnTypeId);
+    List<ProductId> isbns = getDeduplicatedISBNs(productIds, isbnTypeId);
+    isbns.addAll(notISBNs);
+    return isbns;
+  }
+
+  public static  String extractProductId(String value) {
+    if(value == null || !value.contains(" ")) {
+      return value;
+    }
+    return value.substring(0, value.indexOf(" "));
+  }
+
+  public static  String extractQualifier(String value) {
+    if(value == null || !value.contains(" ")) {
```

Comment:
Maybe better to use StringUtils.isNotBlank?

Reply:
Here we have to check if the string contains " ", if so, then we need to return only part of the string. So I think these conditions are normal here.